tests/test_echo.py

```diff
@@ -145,11 +145,15 @@ async def gen():
             yield echo_pb2.EchoRequest(data=data)
 
     assert (await stub.Echo(echo_pb2.EchoRequest(data=data))).data == data
-    assert [response.data for response in await async_iterable_to_list(
-            stub.EchoTwoTimes(echo_pb2.EchoRequest(data=data)))] == [data] * 2
+    async with stub.EchoTwoTimes(echo_pb2.EchoRequest(data=data)) as aiter:
+        assert [response.data for response in await async_iterable_to_list(aiter)] == [
+            data
+        ] * 2
     assert (await stub.EchoLast(gen())).data == data * 4
-    assert [response.data for response in await async_iterable_to_list(
-            stub.EchoEachTime(gen()))] == [data] * 4
+    async with stub.EchoEachTime(gen()) as aiter:
+        assert [response.data for response in await async_iterable_to_list(aiter)] == [
+            data
+        ] * 4
```

Comment:
So there were existing tests that passed a generator-- why didn't they hit the issue you reported?

Just wondering if an additional test is needed, along the lines of the reproduction you submitted.

Reply:
The generators are all iterated to completion, which allows the task group in the generator be cleaned up.  I added with the reproduction, maybe you missed it: https://github.com/python-trio/purerpc/pull/37/files#diff-6e5030d96839d3fe377cf209b6cab5b0d50ae900b458248d8c24b61774117170R81-R95